src/olympia/versions/tests/test_commands.py

```diff
@@ -0,0 +1,329 @@
+
+        for addon in addons_to_ignore_promoted:
+            if hasattr(addon.current_version, '_compatible_apps'):
+                del addon.current_version._compatible_apps
+            assert amo.ANDROID in addon.current_version.compatible_apps
+            assert (
+                addon.current_version.compatible_apps[amo.ANDROID].min.version == '48.0'
+            )
+            assert addon.current_version.compatible_apps[amo.ANDROID].max.version == '*'
+            assert (
+                addon.current_version.compatible_apps[amo.ANDROID].originated_from
+                == amo.APPVERSIONS_ORIGINATED_FROM_UNKNOWN
+            )
+            assert not addon.current_version.file.reload().strict_compatibility
+
+        for addon in addons_to_ignore_not_in_csv:
+            if hasattr(addon.current_version, '_compatible_apps'):
+                del addon.current_version._compatible_apps
+            assert amo.ANDROID not in addon.current_version.compatible_apps
+
+
+class TestForceMaxAndroidCompatibility(TestCase):
+    def setUp(self):
+        self.min_version_fenix = AppVersion.objects.get_or_create(
+            application=amo.ANDROID.id,
+            version=amo.MIN_VERSION_FENIX_GENERAL_AVAILABILITY,
+        )[0]
+        self.max_version_fenix = AppVersion.objects.get_or_create(
+            application=amo.ANDROID.id, version=amo.MAX_VERSION_FENNEC
+        )[0]
+        self.some_fenix_version = AppVersion.objects.get_or_create(
+            application=amo.ANDROID.id, version='99.0'
+        )[0]
+
+    def test_full(self):
+        addons_to_ignore_promoted = [
+            addon_factory(
+                name='Recommended for Android',
+                version_kw={
+                    'application': amo.ANDROID.id,
+                    'min_app_version': '48.0',
+                    'max_app_version': '*',
+                },
+                promoted=RECOMMENDED,
+            ),
+            addon_factory(
+                name='Line for all',
+                version_kw={
+                    'application': amo.ANDROID.id,
+                    'min_app_version': '48.0',
+                    'max_app_version': '*',
+                },
+                promoted=LINE,
+            ),
+        ]
+        addons_to_ignore_119 = [
+            addon_factory(
+                name='Recommended for Android 119',
+                version_kw={
+                    'application': amo.ANDROID.id,
+                    'min_app_version': '119.0a1',
+                    'max_app_version': '*',
+                },
+                promoted=RECOMMENDED,
+            ),
+            addon_factory(
+                name='Normal for Android 119',
+                version_kw={
+                    'application': amo.ANDROID.id,
+                    'min_app_version': '119.0a1',
+                    'max_app_version': '*',
+                },
+                promoted=RECOMMENDED,
+            ),
+        ]
```

Comment:
Can we have some add-ons that are promoted for both Android and Desktop, other promoted groups than Line and Recommended so test they are reset as expected.

Reply:
https://github.com/mozilla/addons-server/pull/21190/commits/ea2458488f142484f59763c679464dea23779af4